src/plugins/dataflow/nodes/factories/transform-rete-node-factory.tsx

```diff
@@ -45,7 +45,12 @@ export class TransformReteNodeFactory extends DataflowReteNodeFactory {
       if (isNaN(n1)) {
         result = NaN;
       } else {
-        result = nodeOperationTypes.method(n1, 0);
+        if (nodeOperationTypes.name === "Ramp") {
+          const recents = (node.data.recentValues as Record<string, any>).nodeValue;
+          result = nodeOperationTypes.method(n1, 0, recents[recents.length - 1]);
+        } else {
+          result = nodeOperationTypes.method(n1, 0);
+        }
```

Comment:
Rather than make a special case of the `Ramp` operation, I think you can simplify it to this along with adding some additional checks:
```suggestion
        const recents = (node.data.recentValues as Record<string, any>).nodeValue;
        const prevValue = Array.isArray(recents) && recents.length > 1 ? recents[recents.length - 1] : undefined;
        result = nodeOperationTypes.method(n1, 0, prevValue);
```
The other `method`s will just ignore the additional argument if provided.

Reply:
That makes sense.  Since ramp is (at the moment) a special case, I was thinking that it made sense to only pass the prevValue when we were using it, but I see now that this also leaves the path open to future functions that may want to access the prior value at the get-go.